src/misc/news/NewsModel.ts

```diff
@@ -0,0 +1,89 @@
+import {createNewsIn, NewsId, NewsOut} from "../../api/entities/tutanota/TypeRefs.js"
+import {IServiceExecutor} from "../../api/common/ServiceRequest.js"
+import {NewsService} from "../../api/entities/tutanota/Services.js"
+import {NotFoundError} from "../../api/common/error/RestError.js"
+import {isOfflineError} from "../../api/common/utils/ErrorCheckUtils.js"
+import {NewsListItem} from "./NewsListItem.js"
+
+
+const newsNameImportMap: Record<string, () => Promise<any>> = {
+	usageOptIn: () => import("./items/UsageOptInNews.js"),
+	test3: () => import("./items/UsageOptInNews.js"), // FIXME remove
+	testOld: () => import("./items/NewsTemplate.js"), // FIXME remove
+	testOlder: () => import("./items/NewsTemplate.js"), // FIXME remove
+	testOldest: () => import("./items/NewsTemplate.js"), // FIXME remove
+}
+
+export class NewsModel {
+	liveNewsIds: NewsId[] = []
+	liveNewsListItems: Record<string, NewsListItem> = {}
+
+	constructor(
+		private readonly serviceExecutor: IServiceExecutor,
+	) {
+	}
+
+	private async importNewsImpl(newsItemName: string) {
+		const importFunc = newsNameImportMap[newsItemName]
+		if (!importFunc) {
+			console.log(`Found no news item implementation for news '${newsItemName}'`)
+			return null
+		} else {
+			const module = await importFunc()
+			return module.default
+		}
+	}
+
+	/**
+	 * Loads the user's unacknowledged NewsItems.
+	 */
+	async loadNewsIds(): Promise<NewsId[]> {
+		try {
+			const response: NewsOut = await this.serviceExecutor.get(NewsService, null)
+
+			this.liveNewsIds = []
+			this.liveNewsListItems = {}
+
+			response.newsItemIds.map(async newsItemId => {
```

Comment:
you are not waiting for this (you just make an array of promises with `map()` and throw it away) and you are not redrawing after loading them either